plugins/BEdita/API/src/Middleware/AnalyticsMiddleware.php

```diff
@@ -0,0 +1,170 @@
+     * @var array
+     */
+    protected $data = [];
+
+    /**
+     * Registered callbacks fo analytics extensions
+     *
+     * @var array
+     */
+    protected static $callbacks = [];
+
+    /**
+     * Configure analytics logger if not configured yet
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->startTime = microtime(true);
+        if (defined('TIME_START')) {
+            $this->startTime = TIME_START;
+        }
+
+        if (!in_array('analytics', Log::configured())) {
+            Log::config('analytics', [
+                'className' => 'File',
+                'path' => LOGS,
+                'scopes' => ['analytics'],
+                'file' => 'analytics',
+            ]);
+        }
+    }
+
+    /**
+     * Getter for $startTime
+     *
+     * @return float
+     * @codeCoverageIgnore
+     */
+    public function getStartTime()
+    {
+        return $this->startTime;
+    }
+
+    /**
+     * Getter for $data
+     *
+     * @return float
+     * @codeCoverageIgnore
+     */
+    public function getData()
+    {
+        return $this->data;
+    }
+
+    /**
+     * Register new analytics extension callback
+     *
+     * @param mixed $newCallback Callback to register
+     * @return void
+     */
+    public static function registerCallback($newCallback)
+    {
+        if (!is_callable($newCallback)) {
+            Log::warning('Bad callback ' . print_r($newCallback, true));
+
+            return;
+        }
+        static::$callbacks[] = $newCallback;
+    }
+
+    /**
+     * Get analytics custom data from registered callbacks
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request The request.
+     * @param \Psr\Http\Message\ResponseInterface $response The response.
+     * @return array
+     */
+    protected function readCallbackData(ServerRequestInterface $request, ResponseInterface $response)
+    {
+        if (empty(static::$callbacks)) {
+            return [];
+        }
+
+        $res = [];
+        $params = [$request, $response];
+        foreach (static::$callbacks as $call) {
+            try {
+                $values = call_user_func_array($call, $params);
+                $res[] = $values;
+            } catch (Throwable $t) {
+                Log::error('Error calling callback ' . print_r($call, true));
+            }
+        }
+
+        return $res;
+    }
+
+    /**
+     * The middleware action.
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request The request.
+     * @param \Psr\Http\Message\ResponseInterface $response The response.
+     * @param callable $next The next middleware to call.
+     * @return \Psr\Http\Message\ResponseInterface A response.
+     */
+    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, callable $next)
+    {
+        $response = $next($request, $response);
+
+        /**
+         * TODO: add custom application error code
+         */
```

Comment:
I suggest to get application error reading the response body just in case of HTTP status code `>= 400`